### PR TITLE
Added customization options for message, member and channel limits when querying channels

### LINKED ISSUE #2840: Update ChannelListViewModel to allow member and message count customization when querying channels
### Context & proposal

We need to pull out member limit query customization into the ViewModel params and update the code.

### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] Allow users to customize the number of messages and members loaded when querying channels.

### PR DESCRIPTION
**Pull request by filbabic**
Fixes #2840 

### 🎯 Goal

We needed to expose a select few properties that are important to our users, when it comes to customizing the query channel requests.

### 🛠 Implementation details

Added three properties - `channelLimit`, `memberLimit` and `messageLimit`.

### 🧪 Testing

You can tinker with the query and the limits to see if the number of items is different when fetching the data.

#### General
- [x] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt [hunk 1]
```diff
@@ -49,6 +49,9 @@ public class ChannelListViewModel(
     public val chatDomain: ChatDomain,
     initialSort: QuerySort<Channel>,
     private val initialFilters: FilterObject,
+    channelLimit: Int = ChannelViewModelFactory.DEFAULT_CHANNEL_LIMIT,
```
Comment (bychkovdmitry) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt:52]: I think `ChannelListViewModel` should be completely unaware of `ChannelViewModelFactory` class. What do you think about leaving the constants in the ViewModel?

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelViewModelFactory.kt [hunk 2]
```diff
@@ -17,11 +17,22 @@ public class ChannelViewModelFactory(
     private val chatDomain: ChatDomain,
     private val querySort: QuerySort<Channel>,
     private val filters: FilterObject,
+    private val channelLimit: Int = DEFAULT_CHANNEL_LIMIT,
```
Comment (MarinTolic) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelViewModelFactory.kt:20]: Missing kdocs here for all parameters.

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt (hunk 1 above)
Comment (MarinTolic) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt:52]: Missing kdocs for newly added parameters.

Comment (filbabic) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt:52]: Agreed! Will push to the VM and then call from the factory!

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessagesViewModelFactory.kt [hunk 3]
```diff
@@ -32,7 +32,7 @@ public class MessagesViewModelFactory(
     private val chatClient: ChatClient = ChatClient.instance(),
     private val chatDomain: ChatDomain = ChatDomain.instance(),
     private val enforceUniqueReactions: Boolean = true,
-    private val messageLimit: Int = 30,
+    private val messageLimit: Int = MessageListViewModel.DEFAULT_MESSAGE_LIMIT,
```
Comment (MarinTolic) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessagesViewModelFactory.kt:35]: We also have `messageLimit: Int = 30` in `MessagesScreen`

**Review by MarinTolic — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] ~~UI Changes correct (before & after images)~~
- [ ] ~~Bugs validated (bugfixes)~~
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessagesViewModelFactory.kt (hunk 3 above)
Comment (filbabic) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/messages/MessagesViewModelFactory.kt:35]: Will update there too - nice catch!

**Review by bychkovdmitry — APPROVED** [review]
LGTM

**Pull request merged**